### ISSUE TITLE
how to change backbone in yolo model pipeline from source code in local env and put this into vision project

### ISSUE THREAD

**Issue by genji970** [ultralytics/ultralytics#20482]
### Search before asking

- [x] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

Hello, i searched for backbone change in yolo model(in specific, yolo v8) and it was hard to figure out how to implement modified yolo v8 without any error.

For example, even if i do something like `model.model[0] = custome_vit`, model's trainer automatically go back to default configuration. So nothing changed.

So, I did, 
1) `git clone [ultralytics](https://github.com/ultralytics/ultralytics)`
2) `pip install ultralytics`
3) read code in local env
4)And solved how to change module in backbone

-> I wish to open my record . 

Second, After I changed a module in backbone in yolo-v8.yaml training time per epoch takes more than when a module is Conv. But model summary does not show the changed one. If I correctly changed module in backbone but summary is wrong, I want to solve this.

### Use case

1) record might be helpful to beginners like me.
2) Easy to get if the model pipeline is change or not.

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @genji970, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

---

🤖 This is an automated response to help you get started. An Ultralytics engineer will review your issue and assist you soon!

**Comment by Y-T-G** [comment]
You need to use the YAML to specify the module.

You can use this PR to define a module and add it to the backbone in YAML.

https://github.com/ultralytics/ultralytics/pull/19615

**Comment by genji970** [comment]
Thank you for your reply. I followed #19615' yaml but still in summary, 

```python
# Parameters
nc: 80
backbone:
  - [-1, 1, SimpleModel, [1, nc]]
head:
  - [0, 1, nn.Identity, []]

module:
  init: |

    import torch.nn as nn

    class SimpleModel(nn.Module):
        def __init__(self, num_classes=10):
            super().__init__()
            self.backbone = nn.Sequential(
                nn.Conv2d(3, 16, kernel_size=3, stride=1, padding=1),  # (3,640,640) -> (16,640,640)
                nn.ReLU(),
                nn.MaxPool2d(2, 2),  # (16,640,640) -> (16,320,320)
                nn.Conv2d(16, 32, kernel_size=3, stride=1, padding=1),  # (16,320,320) -> (32,320,320)
                nn.ReLU(),
                nn.MaxPool2d(2, 2),  # (32,320,320) -> (32,160,160)
                nn.Conv2d(32, 64, kernel_size=3, stride=1, padding=1),  # (32,160,160) -> (64,160,160)
                nn.ReLU(),
                nn.MaxPool2d(2, 2)   # (64,160,160) -> (64,80,80)
            )
            self.head = nn.Sequential(
                nn.Flatten(),  # Flatten (64,80,80) -> (64*80*80)
                nn.Linear(64 * 80 * 80, 128),
                nn.ReLU(),
                nn.Linear(128, num_classes)
            )

        def forward(self, x):
            x = self.backbone(x)
            x = self.head(x)
            return x
  parse: |
    if m is SimpleModel:
      c2 = args[0]
      c1 = ch[f]
      args = [*args[1:]]
```

but still in summary,

```python
                   from  n    params  module                                       arguments                     
  0                  -1  1      1856  ultralytics.nn.modules.conv.Conv             [3, 64, 3, 2]                 
  1                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]               
  2                  -1  3    279808  ultralytics.nn.modules.block.C2f             [128, 128, 3, True]           
  3                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]              
  4                  -1  6   2101248  ultralytics.nn.modules.block.C2f             [256, 256, 6, True]           
  5                  -1  1   1180672  ultralytics.nn.modules.conv.Conv             [256, 512, 3, 2]              
  6                  -1  6   8396800  ultralytics.nn.modules.block.C2f             [512, 512, 6, True]           
  7                  -1  1   2360320  ultralytics.nn.modules.conv.Conv             [512, 512, 3, 2]              
  8                  -1  3   4461568  ultralytics.nn.modules.block.C2f             [512, 512, 3, True]           
  9                  -1  1    656896  ultralytics.nn.modules.block.SPPF            [512, 512, 5]                 
 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 11             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 12                  -1  3   4723712  ultralytics.nn.modules.block.C2f             [1024, 512, 3]                
 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 14             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 15                  -1  3   1247744  ultralytics.nn.modules.block.C2f             [768, 256, 3]                 
 16                  -1  1    590336  ultralytics.nn.modules.conv.Conv             [256, 256, 3, 2]              
 17            [-1, 12]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 18                  -1  3   4592640  ultralytics.nn.modules.block.C2f             [768, 512, 3]                 
 19                  -1  1   2360320  ultralytics.nn.modules.conv.Conv             [512, 512, 3, 2]              
 20             [-1, 9]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 21                  -1  3   4723712  ultralytics.nn.modules.block.C2f             [1024, 512, 3]                
 22        [15, 18, 21]  1   5583571  ultralytics.nn.modules.head.Detect           [1, [256, 512, 512]]     
```
basic conf shows. Is this just my mistake?

**Comment by Y-T-G** [comment]
You're supposed to pass the path to yaml.

```python
model = YOLO("custom_model.yaml")
```

**Comment by genji970** [comment]
now I understand what is the problem thank you

**Comment by glenn-jocher** [comment]
Great to hear you've understood the solution! To summarize for anyone who finds this thread later: when you want to customize a YOLO model architecture, you need to define your custom modules in a YAML file and load it directly with `model = YOLO("custom_model.yaml")` rather than trying to modify model components after initialization. The model parsing system in Ultralytics will then properly build your custom architecture according to the YAML specification.

**Comment by genji970** [comment]
Actually I tried to change source code itself in local env(such as tasks.py,conv.py,utils.__init__.py etc).It was tough(e.g : custome model is training but summary is not right). But now I got easy and precise solution thank you!
